cfme/cloud/instance/__init__.py

```diff
@@ -38,33 +35,116 @@
         'policy_tree': Tree('//div[@class="containerTableStyle"]/table')
     })
 
-image_select_form = Form(
-    fields=[
-        ('template_table', Table('//div[@id="pre_prov_div"]//table')),
-        ('cancel_button', form_buttons.cancel)
-    ]
-)
-
 manage_policies_tree = CheckboxTree("//div[@id='protect_treebox']/ul")
 
 list_table = PagedTable(table_locator="//div[@id='list_grid']//table")
 
 match_page = partial(match_location, controller='vm_cloud', title='Instances')
 
 
-def details_page_check(name, provider):
-    title_match = match_page(summary='Instance "{}"'.format(name))
-    if title_match:
-        # Also check provider
+def details_page_check(provider_name):
         try:
-            prov_match = InfoBlock.text('Relationships', 'Cloud Provider') == provider.name
-            return title_match and prov_match
+            return InfoBlock.text('Relationships', 'Cloud Provider') == provider_name
         except BlockTypeUnknown:
             # Default to false since we can't identify which provider the image belongs to
             return False
 
 
-@VM.register_for_provider_type("cloud")
+class CloudInstanceView(BaseLoggedInPage):
+    title = Text('#explorer_title_text')
+
+    configuration = Dropdown('Configuration')
+    lifecycle = Dropdown('Lifecycle')
+    policy = Dropdown('Policy')
+    power = Dropdown('Power')
+
+    @property
+    def in_cloud_instance(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Clouds', 'Instances']
+        )
+
+    @View.nested
+    class instances_by_provider(Accordion): # noqa
+        ACCORDION_NAME = 'Instances by Provider'
+        tree = ManageIQTree()
+
+    @View.nested
+    class images_by_provider(Accordion): # noqa
+        ACCORDION_NAME = 'Images by Provider'
+        tree = ManageIQTree()
+
+    @View.nested
+    class instances(Accordion): # noqa
+        ACCORDION_NAME = 'Instances'
+        tree = ManageIQTree()
+
+    @View.nested
+    class images(Accordion): # noqa
+        ACCORDION_NAME = 'Images'
```

Comment:
I see - wasn't sure about the more_complex_names, and how they would be parsed/matched. Personally I like setting it explicitly, its more clear to me (at my current level of proficiency) what the nested view class is for.  Not against removing if we want to keep a consistent style.